# Scripts description 


We followed the [ipyrad_script](https://github.com/daniel-acga/RADSeqMicrobiome/blob/master/bin/ipyrad_script) which corresponds to the default pipeline from the program modifying the next parameters from the [params-.txt file](https://github.com/daniel-acga/RADSeqMicrobiome/blob/master/bin/params-.txt):


1. clustering treeshold: 0,97 and 0,95 
1. min_samples_locus: 1, chosen so as not to discard the sequences from possible exclusive symbionts of a particular sample
1. max_alleles_consens: 4 to prevent the influence of allelic differences for identifying a differences between microbial sequences.


Then, for each strategy, the .loci file generated by ipyrad, that presents the reads associated with each locus, was saved with the name of the individual from which the sample originated. We used the R script, [holmes_sp.R](https://github.com/daniel-acga/RADSeqMicrobiome/blob/master/bin/holmes_sp.R) modified from [Holmes and Rabosky, 2018](https://peerj.com/articles/4662/) to take the first sequence of each locus and group them into a fasta file that was later processed using the [blastn tool](https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastDocs&DOC_TYPE=Download
) from the National Center for Biotechnology Information (NCBI).

After the blast was made we finished using Holmes and Rabosky's script to retrieve the blast matches with at least 90% of similarity and to make diversity inferences.
We compared the association from different taxa to their script to count the loci from different RADSeq samples called [count_loci.R](https://github.com/daniel-acga/RADSeqMicrobiome/blob/master/bin/count_loci.R)
